src/odemis/acq/align/shift.py

```diff
@@ -19,167 +19,158 @@
+    upsample_factor (int, optional): The upsampling factor.  Defaults to 1.
+    axis_offsets (tuple of int, optional): The offsets of the region to be sampled. Defaults to None (uses
+      image center)
+
+    returns (numpy.ndarray): The upsampled DFT of the specified region.
+    """
+    # if people pass in an integer, expand it to a list of equal-sized sections
+    if not hasattr(upsampled_region_size, "__iter__"):
+        upsampled_region_size = [upsampled_region_size, ] * data.ndim
+    else:
+        if len(upsampled_region_size) != data.ndim:
+            raise ValueError("shape of upsampled region sizes must be equal "
+                             "to input data's number of dimensions.")
+
+    if axis_offsets is None:
+        axis_offsets = [0, ] * data.ndim
+    else:
+        if len(axis_offsets) != data.ndim:
+            raise ValueError("number of axis offsets must be equal to input "
+                             "data's number of dimensions.")
+
+    im2pi = 1j * 2 * numpy.pi
+
+    dim_properties = list(zip(data.shape, upsampled_region_size, axis_offsets))
+
+    for (n_items, ups_size, ax_offset) in dim_properties[::-1]:
+        kernel = ((numpy.arange(ups_size) - ax_offset)[:, None]
+                  * fftfreq(n_items, upsample_factor))
+        kernel = numpy.exp(-im2pi * kernel)
+        # use kernel with same precision as the data
+        kernel = kernel.astype(data.dtype, copy=False)
+
+        # Equivalent to:
+        #   data[i, j, k] = kernel[i, :] @ data[j, k].T
+        data = numpy.tensordot(kernel, data, axes=(1, -1))
+    return data
+
 
 def MeasureShift(previous_img, current_img, precision=1):
     """
-    Given two images, it calculates the shift in x and y axis. It first computes
-    the cross-correlation of the two images and then locates the peak. The coordinates
-    of the peak of the cross-correlation define the shift vector between the two images.
-    This implementation is based on the "Efficient subpixel image registration by
+    The function is taken from skimage.registration._phase_cross_correlation.
+    Given two images, it calculates the shift in x and y axis. It first computes the cross-correlation of the
+    two images. Then it modifies cross-correlation values
+    by reducing the amplitude of noise and enhancing the shift amplitude. The coordinates
+    of the maximum of the cross-correlation define the shift vector between the two images.
+    The implementation of precision > 1 is based on the "Efficient subpixel image registration by
     cross-correlation" by Manuel Guizar, for the corresponding matlab code see
     http://www.mathworks.com/matlabcentral/fileexchange/
     18401-efficient-subpixel-image-registration-by-cross-correlation.
 
+    For e.g.
+    When the current image moves to top right corner relatively to the previous image.
+    The returned shift value is positive in both x and y respectively.
```

Comment:
Are you sure about this? Normally the convention is that X,Y positive (in pixels) mean a move towards *bottom* right. I looked quickly at the caller code and it seems to follow that convention...

Reply:
Sorry, When the current image moves to **top left corner** relatively to the previous image.
    The returned shift value is positive in both x and y respectively. It is when object in the second image moves to top left corner w.r.t to first image, shift in col and row is measured in positive sign. Do you see the same way?